src/server/protocols/v1/websocket.rs

```diff
@@ -0,0 +1,281 @@
+        if self.clients.is_empty() {
+            return;
+        }
+
+        let string = serde_json::to_string_pretty(value).unwrap();
+        for client in &self.clients {
+            let is_match = client.re.as_ref().map_or(false, |regx| regx.is_match(name));
+            if is_match {
+                client.actor.do_send(StringMessage(string.clone()));
+            }
+        }
+    }
+    pub fn get_client_count(&self) -> usize {
+        self.clients.len()
+    }
+}
+
+lazy_static! {
+    static ref MANAGER: Arc<Mutex<WebsocketManager>> =
+        Arc::new(Mutex::new(WebsocketManager::default()));
+}
+
+pub fn send_to_websockets(message: Value) {
+    MANAGER.lock().unwrap().send(&message, &message.to_string());
+}
+
+#[derive(Debug)]
+pub struct WebsocketActor {
+    server: Arc<Mutex<WebsocketManager>>,
+    pub filter: String,
+}
+
+impl WebsocketActor {
+    pub fn new(message_filter: String) -> Self {
+        Self {
+            server: MANAGER.clone(),
+            filter: message_filter,
+        }
+    }
+}
+
+impl Handler<StringMessage> for WebsocketActor {
+    type Result = ();
+
+    fn handle(&mut self, message: StringMessage, context: &mut Self::Context) {
+        context.text(message.0);
+    }
+}
+
+impl Actor for WebsocketActor {
+    type Context = ws::WebsocketContext<Self>;
+}
+
+impl StreamHandler<Result<ws::Message, ws::ProtocolError>> for WebsocketActor {
+    fn started(&mut self, ctx: &mut Self::Context) {
+        log::info!("Starting websocket, add itself in manager.");
+        self.server
+            .lock()
+            .unwrap()
+            .clients
+            .push(WebsocketActorContent {
+                actor: ctx.address(),
+                re: Regex::new(&self.filter).ok(),
+            });
+    }
+
+    fn finished(&mut self, ctx: &mut Self::Context) {
+        log::info!("Finishing websocket, remove itself from manager.");
+        self.server
+            .lock()
+            .unwrap()
+            .clients
+            .retain(|x| x.actor != ctx.address());
+    }
+
+    fn handle(&mut self, msg: Result<ws::Message, ws::ProtocolError>, ctx: &mut Self::Context) {
+        match msg {
+            Ok(ws::Message::Ping(msg)) => ctx.pong(&msg),
+            Ok(ws::Message::Text(text)) => {
+                let v: Vec<&str> = text.split("&&").collect();
+
+                for request in v {
+                    let request = request.trim();
+                    if request.starts_with('/') {
+                        ctx.text(request_endpoint(request));
+                    } else {
+                        let error_msg = format!(
+                            "{} {}, missing / ?",
+                            json!("Error: Invalid command:"),
+                            request
+                        );
+                        ctx.text(error_msg)
+                    }
+                }
+            }
+            Ok(ws::Message::Binary(bin)) => ctx.binary(bin),
+            _ => (),
```

Comment:
Shouldn't we handle the Close frame?

Reply:
@joaoantoniocardoso Will test it:
`Ok(ws::Message::Close(msg)) => ctx.close(msg)`

If it works we can add to mavlink2rest.

Reply:
![image](https://github.com/bluerobotics/navigator-web-assistant/assets/80598030/8014d0b8-dc6b-4fa4-b566-4ad03ab1e554)
@joaoantoniocardoso It works, thanks, before this the connection keep holding for some seconds or minutes until it really closes and finish session on websocket client, we need to add this do mavlink2rest.